tutorials/io/ntuple/ntpl015_processor_join.C

```diff
@@ -77,23 +77,25 @@ void Read()
    TH1F hPy("h", "This is the px + py distribution", 100, -4, 4);
    hPy.SetFillColor(48);
 
-   // The first specified ntuple is the main ntuple and will be used to drive the processor loop. The subsequent
-   // list of ntuples (in this case, only one) are auxiliary and will be joined with the entries from the main ntuple.
-   // We specify field "i" as the join field. This field, which should be present in all ntuples specified is used to
-   // identify which entries belong together. Multiple join fields can be specified, in which case the combination of
-   // field values is used. It is possible to specify up to 4 join fields. Providing an empty list of join fields
-   // signals to the processor that all entries are aligned.
+   // The first specified ntuple is the primary ntuple and will be used to drive the processor loop. The subsequent
+   // list of ntuples (in this case, only one) are auxiliary and will be joined with the entries from the primary
+   // ntuple. We specify field "i" as the join field. This field, which should be present in all ntuples specified is
+   // used to identify which entries belong together. Multiple join fields can be specified, in which case the
+   // combination of field values is used. It is possible to specify up to 4 join fields. Providing an empty list of
+   // join fields signals to the processor that all entries are aligned.
    auto processor =
-      RNTupleProcessor::CreateJoin({kMainNTupleName, kMainNTuplePath}, {kAuxNTupleName, kAuxNTuplePath}, {"i"});
-
-   float px, py;
-   for (const auto &entry : *processor) {
-      // Fields from the main ntuple are accessed by their original name.
-      px = *entry.GetPtr<float>("vpx");
-      // Fields from auxiliary ntuples are accessed by prepending the name of the auxiliary ntuple.
-      py = *entry.GetPtr<float>(kAuxNTupleName + ".vpy");
-
-      hPy.Fill(px + py);
+      RNTupleProcessor::CreateJoin({kPrimaryNTupleName, kMainNTuplePath}, {kAuxNTupleName, kAuxNTuplePath}, {"i"});
+
+   // Access to the processor's fields is done by first registering them through RNTupleProcessor::RegisterField(). The
+   // returned value can be used to read the current entry's value for that particular field. Fields from the primary
+   // ntuple are registered by their original name.
+   auto px = processor->RegisterField<float>("vpx");
```

Comment:
```suggestion
   auto px = processor->RequestField<float>("vpx");
```
which may (or may not) closer to the actual semantic of the function.